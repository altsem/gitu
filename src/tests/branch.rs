use super::*;

fn setup() -> TestContext {
    let ctx = TestContext::setup_clone();
    run(ctx.dir.path(), &["git", "checkout", "-b", "merged"]);
    run(ctx.dir.path(), &["git", "checkout", "-b", "unmerged"]);
    commit(ctx.dir.path(), "first commit", "");
    run(ctx.dir.path(), &["git", "checkout", "main"]);
    ctx
}

#[test]
fn branch_menu() {
    snapshot!(setup(), "Yjb");
}

#[test]
fn switch_branch_selected() {
    snapshot!(setup(), "Yjjbb<enter>");
}

#[test]
fn switch_branch_input() {
    snapshot!(setup(), "Ybbmerged<enter>");
}

#[test]
fn checkout_new_branch() {
    snapshot!(setup(), "bcnew<enter>");
}

#[test]
fn delete_branch_selected() {
    snapshot!(setup(), "YjjbK<enter>");
}

#[test]
fn delete_branch_input() {
    snapshot!(setup(), "bKmerged<enter>");
}

#[test]
fn delete_branch_empty() {
    snapshot!(setup(), "bK<enter>");
}

#[test]
fn delete_unmerged_branch() {
    // TODO: Remove <esc> once #368 is fixed
    snapshot!(setup(), "bKunmerged<enter>n<esc>bKunmerged<enter>y");
}
