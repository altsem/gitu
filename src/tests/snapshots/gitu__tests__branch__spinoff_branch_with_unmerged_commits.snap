---
source: src/tests/branch.rs
expression: ctx.redact_buffer()
---
▌On branch new                                                                  |
                                                                                |
 Recent commits                                                                 |
 c84f226 new add first commit                                                   |
 b66a0bf main origin/main add initial-file                                      |
                                                                                |
                                                                                |
                                                                                |
────────────────────────────────────────────────────────────────────────────────|
Branch                                                                          |
b Checkout branch/revision                                                      |
c Checkout new branch                                                           |
s Spinoff branch                                                                |
K Delete branch                                                                 |
q/esc Quit/Close                                                                |
────────────────────────────────────────────────────────────────────────────────|
$ git checkout -b new                                                           |
Switched to a new branch 'new'                                                  |
$ git update-ref -m "reset: moving to b66a0bf82020d6a386e94d0fceedec1f817d20c7" |
> Branch main was reset to b66a0bf82020d6a386e94d0fceedec1f817d20c7             |
styles_hash: 10f4e2a83bdba319
